### PR TITLE
indiv piece models

### PR DESCRIPTION
**Pull request by mce777**
Created The individual piece models: king, queen, rook, bishop, knight, pawn....Currently no columns within these models. They inherit from the piece model.

### CODE REVIEW

**Comment by KaoruDev** [comment]
WAIT!!!

it looks like you created the models via `rails generate model` which caused it to _also_ create migrations. Don't think we need the migrations eh? Also the schema has been modified.

**Comment by mce777** [comment]
so you don't  run migrations for the indiv pieces?

**Comment by KaoruDev** [comment]
No so the `King`, `Queen` etc models represent records in the `Pieces` table with the `King`, `Queen` type respectively. That == `Single` in STI

**Comment by KaoruDev** [comment]
oh and so you're also missing the `type` column in the `pieces` table ;)

**Comment by mce777** [comment]
ok, got it. No new migrations necessary for the sub pieces. But they can be separate files.

Type column is already in the pieces table :) I'll delete this branch and redo the sub models.

**Comment by KaoruDev** [comment]
you don't have to delete this branch, unless you already have?, just gotta delete those migrations. as for the schema, remember how you fixed that merge conflict? same thing, cause it will only re-run the migration files you have.

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class Piece < ActiveRecord::Base
+  belongs_to :game
+
+  def self.types
+    %w(King Queen Rook Bishop Knight Pawn)
```
Comment (KaoruDev) [inline app/models/piece.rb:5]: not sure what this is used for?

Comment (mce777) [inline app/models/piece.rb:5]: got it from that link you providided in trello: http://samurails.com/tutorial/single-table-inheritance-with-rails-4-part-1/

seemed to make sense.

**Comment by mce777** [comment]
thanks for that tip. If it wasn't so long, "rake db:drop db:create db:migrate" would be my new tattoo.

File: app/models/piece.rb (hunk 1 above)
Comment (KaoruDev) [inline app/models/piece.rb:5]: what's the TLDR?

**Comment by KaoruDev** [comment]
ha you can always setup a `bash` alias, but that's beyond the scope of this project

File: app/models/piece.rb (hunk 1 above)
Comment (mce777) [inline app/models/piece.rb:5]:
```
# We will need a way to know which animals
# will subclass the Animal model
def self.races
  %w(Lion WildBoar Meerkat)
end
```

FYI for whatever reason he replaced "type" with "races".

Comment (KaoruDev) [inline app/models/piece.rb:5]: yep, does that apply to us? I don't see where `Animals.races` is being used.

**Pull request merged**